libdnf5-plugins/local/local.cpp

```diff
@@ -0,0 +1,247 @@
+// Copyright Contributors to the DNF5 project
+// SPDX-License-Identifier: THE-LICENSE-IDENTIFIER
+
+#include "utils/string.hpp"
+
+#include "libdnf5/utils/bgettext/bgettext-mark-domain.h"
+
+#include <fcntl.h>
+#include <libdnf5/base/base.hpp>
+#include <libdnf5/conf/const.hpp>
+#include <libdnf5/plugin/iplugin.hpp>
+#include <libdnf5/repo/repo.hpp>
+#include <libdnf5/repo/repo_query.hpp>
+#include <sys/wait.h>
+
+
+constexpr const char * LOCAL_REPO_NAME{"_dnf_local"};
+constexpr const char * LOCATION_IN_PERSISTDIR{"plugins/local"};
+
+using namespace libdnf5;
+
+namespace {
+
+constexpr const char * PLUGIN_NAME{"local"};
+constexpr libdnf5::plugin::Version PLUGIN_VERSION{.major = 1, .minor = 0, .micro = 0};
+constexpr PluginAPIVersion REQUIRED_PLUGIN_API_VERSION{.major = 2, .minor = 0};
+
+constexpr const char * attrs[]{"author.name", "author.email", "description", nullptr};
+constexpr const char * attrs_value[]{
+    "Aleš Matěj",
+    "amatej@redhat.com",
+    "Automatically copy all downloaded packages to a repository configured path and generate repo metadata."};
+
+
+class LocalPluginError : public libdnf5::Error {
+    using Error::Error;
+    const char * get_domain_name() const noexcept override { return "libdnf5::plugin"; }
+    const char * get_name() const noexcept override { return "LocalPluginError"; }
+};
+
+
+class LocalPlugin : public plugin::IPlugin {
+public:
+    LocalPlugin(libdnf5::plugin::IPluginData & data, libdnf5::ConfigParser & config) : IPlugin(data), config(config) {}
+    virtual ~LocalPlugin() = default;
+
+    PluginAPIVersion get_api_version() const noexcept override { return REQUIRED_PLUGIN_API_VERSION; }
+
+    const char * get_name() const noexcept override { return PLUGIN_NAME; }
+
+    plugin::Version get_version() const noexcept override { return PLUGIN_VERSION; }
+
+    const char * const * get_attributes() const noexcept override { return attrs; }
+
+    const char * get_attribute(const char * attribute) const noexcept override {
+        for (size_t i = 0; attrs[i]; ++i) {
+            if (std::strcmp(attribute, attrs[i]) == 0) {
+                return attrs_value[i];
+            }
+        }
+        return nullptr;
+    }
+
+    void post_base_setup() override {
+        Base & base = get_base();
+        auto repo_sack = base.get_repo_sack();
+        auto repo = repo_sack->create_repo(LOCAL_REPO_NAME);
+        if (config.has_option("main", "repodir")) {
+            repodir = config.get_value("main", "repodir");
+        } else {
+            // default repo location
+            repodir = base.get_config().get_persistdir_option().get_value();
+            repodir = repodir / LOCATION_IN_PERSISTDIR;
+        }
+        repo->get_config().get_name_option().set("Local libdnf5 plugin repo");
+        repo->get_config().get_baseurl_option().set("files://" / repodir);
```

Comment:
The correct scheme is `file://`. This works since `a / b` where `b` is an absolute path will evaluate to just `b`. But I think the following would be more correct:

```
repo->get_config().get_baseurl_option().set("file://" + std::filesystem::absolute(repodir).string());
```

Reply:
Fixed, thanks!